Library/Homebrew/rubocops/urls_cop.rb

```diff
@@ -0,0 +1,102 @@
+require_relative "./extend/formula_cop"
+
+module RuboCop
+  module Cop
+    module FormulaAudit
+      # This cop audits urls and mirrors in Formulae
+      class Urls < FormulaCop
+        def audit_formula(_node, _class_node, _parent_class_node, body_node)
+          #### All patterns would be defined here
+
+          # GNU urls; doesn't apply to mirrors
+          gnu_pattern = %r{^(?:https?|ftp)://ftpmirror.gnu.org/(.*)}
+
+          # Fossies upstream requests they aren't used as primary URLs
+          # https://github.com/Homebrew/homebrew-core/issues/14486#issuecomment-307753234
+          fossies_pattern = %r{^https?://fossies\.org/}
+
+          # Check a variety of SSL/TLS URLs that don't consistently auto-redirect
+          # or are overly common errors that need to be reduced & fixed over time.
+          http_urls_pattern = Regexp.union([%r{^http://ftp\.gnu\.org/},
```

Comment:
Perhaps this could use a name other than `http_urls_pattern`? One that better reflects the fact that the urls which match this pattern should be changed to `https://` might be a better fit . `https_url_id_pattern` is a bad substitute, and but it would probably be good to make this rely a little less on the inline comment. However, if nothing better comes to mind this is still fine. 😄